client/src/includes/contentMetrics.ts

```diff
@@ -0,0 +1,86 @@
+interface ContentMetrics {
+  wordCount: number;
+  readingTime: number;
+}
+
+interface SegmentData {
+  segment: string;
+  isWordLike?: boolean | undefined;
+}
+
+export const getContentMetrics = (
+  lang: string,
+  text: string,
+): ContentMetrics => {
+  let wordCount = 0;
+
+  if (typeof Intl.Segmenter === 'function') {
+    const segmenter = new Intl.Segmenter(lang, { granularity: 'word' });
+    const segments: SegmentData[] = Array.from(segmenter.segment(text));
+    wordCount = segments.reduce(
+      (count, segment) => (segment.isWordLike ? count + 1 : count),
+      0,
+    );
+  } else {
+    // Fallback to regex if Intl.Segmenter is not supported
+    wordCount =
+      text
+        .trim()
+        .replace(/['";:,.?¿\-!¡]+/g, '')
+        .match(/\S+/g)?.length || 0;
+  }
+
+  // Silent-reading adults average 238 words per minute
```

Comment:
See my comment on https://github.com/wagtail/wagtail/issues/12020#issuecomment-2175775263 – I think this needs further work to use wpm per language, and a fallback to 238 if we don’t have a language-specific number.

We should also have quite a bit of code comments here to explain how we arrived at those numbers, with a link directly to the study’s pre-print: https://osf.io/preprints/psyarxiv/xynwg/, and the DOI `https://doi.org/10.1016/j.jml.2019.104047`.